### ISSUE TITLE
Dart 3.0: Code crashes when query is inside condition with "code removed by Dart AOT compiler"

### ISSUE THREAD

**Issue by bananowysong** [objectbox/objectbox-dart#555]
Description: Code crashes when query is inside condition with "code removed by Dart AOT compiler"

### Basic info:

- objectbox 2.1.0
- Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.2, on macOS 13.2.1 22D68 darwin-arm64, locale en-en)
    • Flutter version 3.13.2 on channel stable at 
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ff5b5b5fa6 (8 days ago), 2023-08-24 08:12:28 -0500
    • Engine revision b20183e040
    • Dart version 3.1.0
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at
    • Platform android-33, build-tools 32.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.81.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.70.0

[✓] Connected device (5 available)
   
[✓] Network resources
    • All expected network resources are available.

- Reproducible: [always]
- Build OS: [macOs ]
- Deployment device or OS: [android emulator/android device galaxy a52s/iphone xs max ]

### Steps to reproduce

_TODO_

1. Put 
```
 
    final Query<Whatever> query;
    if (condition) {
      query = dataBox
          .query(Model_.value.equals(id))
          .build();
    } else {
      query = dataBox.query(Model_.otherValue.equals(id)).build();
    }

    final data = query.find();
    query.close();
    return data;
  }
```
3.
In debug it will work, but in release version this code will crash, with:
"Attempt to execute code removed by Dart AOT compiler".
The function does not run at all. 

### Expected behavior
It would not crash and return data. 

### Additional context

Started after upgrading flutter to 3.13.2 from 3.1.

- May transactions or multi-threading play a role? Likely not.
- Did you find any workarounds to prevent the issue? Yeah, 
```
final data;
     if (condition) {
final Query<Whatever> query = dataBox
          .query(Model_.value.equals(id))
          .build();
data = query.find();
    query.close();
    } else {
      final Query<Whatever> query = dataBox.query(Model_.otherValue.equals(id)).build();
data = query.find();
    query.close();
    }
    return data;
```

**Comment by greenrobot-team** [comment]
Thanks for reporting!

Do you have a stack trace of the crash? Does this also occur with the latest version of ObjectBox?

Maybe we can make use of [some pragma annotations](https://mrale.ph/dartvm/compiler/aot/entry_point_pragma.html) (though I don't see how that one would help here). Or this is a Dart bug which we should then report.

**Comment by greenrobot-team** [comment]
Confirmed this also occurs with the latest version. Could reproduce with our Flutter example.

**Comment by greenrobot-team** [comment]
[This is a known Dart issue](https://github.com/dart-lang/sdk/issues/52596) which should be fixed with a future Dart release (likely 3.2.0). See also a [related issue for workarounds](https://github.com/dart-lang/sdk/issues/52874).

@bananowysong An alternative to your workaround  is to make `query` non-final and nullable, e.g.
```dart
Query<Whatever>? query;
if (condition) {
  query = dataBox
      .query(Model_.value.equals(id))
      .build();
} else {
  query = dataBox.query(Model_.otherValue.equals(id)).build();
}
final data = query!.find();
query!.close();
```
